web/backend/src/controllers/users.ts

```diff
@@ -34,17 +34,29 @@ usersRouter.post('/add_role', (req, res, next) => {
     }
   }
 
-  addPolicy(req.body.userId, req.body.subject, req.body.permission)
-    .then(() => {
-      res.set(200).send();
-      next();
-    })
-    .catch((e) => {
-      res.status(400).send(`Error while adding to roles: ${e}`);
-    });
-
-  // Call https://search-api.epfl.ch/api/ldap?q=228271, if the answer is
-  // empty then sciper unknown, otherwise add it in userDB
+  if ('userId' in req.body) {
+    try {
+      addPolicy(req.body.userId, req.body.subject, req.body.permission);
```

Comment:
```suggestion
      await addPolicy(req.body.userId, req.body.subject, req.body.permission);
```

And you'll also need to add an `async` somewhere in the definition of the callback, IIRC, like that:

```
usersRouter.post('/add_role', async (req, res, next) => {
```

Reply:
used `Promise.resolve` because I wasn't sure about the async in the callback, does that make sense?